lib/IRGen/IRGen.cpp

```diff
@@ -1069,32 +1076,36 @@ static void performParallelIRGeneration(IRGenOptions &Opts,
   }
 }
 
-
 std::unique_ptr<llvm::Module> swift::
 performIRGeneration(IRGenOptions &Opts, swift::ModuleDecl *M,
                     std::unique_ptr<SILModule> SILMod,
-                    StringRef ModuleName, llvm::LLVMContext &LLVMContext,
+                    StringRef ModuleName, const PrimarySpecificPaths &PSPs,
+                    llvm::LLVMContext &LLVMContext,
+                    ArrayRef<std::string> parallelOutputFilenames,
                     llvm::GlobalVariable **outModuleHash) {
-  if (SILMod->getOptions().shouldPerformIRGenerationInParallel()) {
+  if (SILMod->getOptions().shouldPerformIRGenerationInParallel() &&
+      !parallelOutputFilenames.empty()) {
```

Comment:
Worth asserting instead of testing?

Reply:
I'll try it, but the callers that invoke this for REPL (for instance), may fail if NumThreads is set.

Reply:
It may be possible to fail the assertion if the command line sets -num-threads but calls for immediate execution. I could modify the callers on that path to pass in the right file names. What do you think?

Reply:
Ah, interesting. All right, I guess it makes sense as is.

Reply:
OK, thanks. Will keep it as is.